python/lsst/ip/isr/isrTask.py

```diff
@@ -831,9 +838,28 @@ def runQuantum(self, butlerQC, inputRefs, outputRefs):
                 if not isinstance(inputs["defects"], Defects):
                     inputs["defects"] = Defects.fromTable(inputs["defects"])
 
-        # TODO: DM-22776 add retrieval for brighter-fatter kernel for Gen3
-        # if we can get HSC to use a new kernel, or just translate the old
-        # one to the new format we could drop the whole new/old style support
+        # Load the correct style of brighter fatter kernel, and repack
+        # the information as a numpy array.
+        if self.config.doBrighterFatter:
+            brighterFatterKernel = inputs.pop('newBFKernel', None)
+            if brighterFatterKernel is None:
+                brighterFatterKernel = inputs.get('bfKernel', None)
+
+            if brighterFatterKernel is not None and not isinstance(brighterFatterKernel, numpy.ndarray):
+                detId = detector.getId()
+                inputs['bfGains'] = brighterFatterKernel.gain
```

Comment:
bfGains does not seem to be populated if not newBFKernel, is that what you want?

Reply:
That is correct.  bfGains will not be in inputs, will then be None inside run(), and will be passed as such to the correction code (which handles the None case by using camera gain values).